SH.Analytics/SH.Analytics.Web/Services/Classes/ApplicationEventService.cs

```
@@ -0,0 +1,161 @@
+                purchaseDetails.ApplicationEventId = entity.Id;
+                purchaseDetails.Amount = model.PurchaseDetails.Amount;
+                purchaseDetails.ProductTitle = model.PurchaseDetails.ProductTitle;
+                await purchaseDetailsRepository.Add(purchaseDetails);
+            }
+        }
+
+        public async Task SaveBulkAsync(List<NewEventModel> models, ApplicationTypes application)
+        {
+            try
+            {
+                foreach (var model in models)
+                {
+                    var entity = new ApplicationEvent();
+                    entity.Application = application;
+                    entity.ApplicationUserId = model.UserId;
+                    entity.ApplicationUsername = model.Username;
+                    entity.Type = model.Type;
+                    entity.CreatedDate = model.CreateDate.HasValue ? model.CreateDate.Value : entity.CreatedDate;
+                    var deviceDetails = new DeviceDetails();
+                    deviceDetails.Device = model.DeviceDetails.Device;
+                    deviceDetails.OSVersion = model.DeviceDetails.OSVersion;
+                    var deviceId = await deviceDetailsRepository.Add(deviceDetails);
+                    entity.DeviceDetailsId = deviceId.Id;
+                    await applicationEventsRepository.Add(entity);
+
+                    if (entity.Type == EventTypes.Purchase)
+                    {
+                        var purchaseDetails = new PurchaseDetails();
+                        purchaseDetails.ApplicationEventId = entity.Id;
+                        purchaseDetails.Amount = model.PurchaseDetails.Amount;
+                        purchaseDetails.ProductTitle = model.PurchaseDetails.ProductTitle;
+                        await purchaseDetailsRepository.Add(purchaseDetails);
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                var a = e;
```

never ignore exceptions